src/mish2jsonl/mish2jsonl.c

```diff
@@ -0,0 +1,160 @@
+#include <stdio.h>
+#include <stdint.h>
+#include <stdlib.h>
+#include <string.h>
+#include "mish_common.h"
+
+typedef struct m_string {
+  uint64_t len;
+  char *string;
+} m_string;
+
+typedef struct worker_output {
+  uint32_t status;   // 4
+  uint16_t ndecoded; // 2
+  uint32_t workerno; // 4
+  m_string workerso; // 16
+  m_string result;   // 16
+} worker_output;
+
+typedef struct cohort_results {
+  uint32_t nworkers;
+  m_string input;
+  worker_output *outputs;
+} cohort_results;
+
+cohort_results results;
+FILE *m_parsing_file;
+int m_finished_parsing;
+
+void m_cohort_print_json(FILE *f, cohort_results *r) {
+  fprintf(f, "{ \"nworkers\": %u, \"input\": \"%s\", \"outputs\": [", r->nworkers, r->input.string);
+  for (int i = 0; i < r->nworkers; i++) {
+    if (i != 0) {
+      fprintf(f, ",");
+    }
+    fprintf(f,
+            "{ \"status\": { \"value\": %u, \"name\": \"%s\" }, \"ndecoded\": %u, \"workerno\": "
+            "%u, \"worker_so\": \"%s\",\"len\": %ld, \"result\": \"%s\" }",
+            r->outputs[i].status, status2str(r->outputs[i].status), r->outputs[i].ndecoded,
+            r->outputs[i].workerno, r->outputs[i].workerso.string, r->outputs[i].result.len,
+            r->outputs[i].status == 1 ? r->outputs[i].result.string : "bad");
```

Comment:
Nit: Don't emit a special `"bad"` string here -- let's just dump whatever's in `result.string`, which *should* be empty (or whatever the decoder puts on errors) on decoding errors.